src/snakemake/scheduling/greedy.py

```
@@ -0,0 +1,135 @@
+from dataclasses import dataclass, field
+from itertools import filterfalse
```

Ruff (0.12.2)

`itertools.filterfalse` imported but unused

Remove unused import: `itertools.filterfalse`

(F401)